aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py

```diff
@@ -270,9 +272,13 @@ async def _get_suite_for_detail(
         )
 
         did_info = await self._did_info_for_did(issuer_id)
-        verification_method = verification_method or self._get_verification_method(
-            issuer_id
-        )
+        verkey_id_strategy = self.profile.context.inject(DefaultVerificationKeyStrategy)
+        verification_method = verkey_id_strategy.get_verkey_id_for_did(issuer_id)
+
+        if verification_method is None:
+            raise V20CredFormatError(
+                f"Unable to get retrieve verification method for did {issuer_id}"
+            )
```

Comment:
Thinking about it a bit more, If we want to make this a generic / pluggable interface, I think the method should have more input so you can correctly define the verification method.

It should contain:
 - proof_type (or something like `allowed_verification_method_types`)
 - optional proof_purpose (so we only return a verification method that is allowed to be used for the intended purpose (assertionMethod, authentication, keyAgreement, etc..)
 - did

Also, maybe it should return an array of verification method instances itself, so the caller can determine based on the whole verification method object + can choose which one to take if there's multiple candidates (currnet callees could just take the first one)

Reply:
Hi @TimoGlastra, thanks for your review.

> I think making this pluggable could work, however maybe it'd be simpler to just provide the `verification_method` as part of the issue credential options?

Indeed, I inadvertently removed this part. I added this back again, so now during issuing it'll try to use the given `verification_method` (if provided) and will otherwise default to generating it dynamically. However, for other protocols such as credential presentation, I think we still need a way to dynamically produce the `verification_method` ID, as the ACA-Py instance of the presenter could be set to automatically accept presentation requests, in which cases the ID cannot be given in the options.

> I would also suggest renaming it to `get_verification_method_id_for_did`.

Done

> Also, maybe it should return an array of verification method instances itself, so the caller can determine based on the whole verification method object + can choose which one to take if there's multiple candidates (currnet callees could just take the first one)

I'm not sure I understand this: from my limited understanding of how ACA-Py works, only a single verification method is used at a time and is stored in the `verkey` field of a `DIDInfo` record returned by using `get_local_did`, or others. This pluggable method is a way to produce the verification method ID associated with this `verkey`, as ACA-Py doesn't store it in this `DIDInfo` record. Are you suggesting that we remove this `verkey` field, and instead use this newly added class to query for a verification method? And then it would return an array of verification method instances with each verification method instance containing a `verkey` plus an ID?

Reply:
On your last point. Yeah what I was saying doesn't fully make sense, as we already have the public key (verkey) and the did, we just want to know the verification method id (so did + `#` part). I was thinking more of determining the best 

However, I think we should provide more than just the `issuer_id`/`did` to determine the `verification_method_id`. In ACA-Py's current model dids only have a single key, but as we already know the public key, the proof type, etc.. I think it would be good to pass it, so once ACA-Py starts supporitng multiple keys for a did, this interface won't have to change.

Reply:
> However, I think we should provide more than just the issuer_id/did to determine the verification_method_id. In ACA-Py's current model dids only have a single key, but as we already know the public key, the proof type, etc.. I think it would be good to pass it, so once ACA-Py starts supporitng multiple keys for a did, this interface won't have to change.

Done